src/Microsoft.AspNetCore.Razor.Evolution/Legacy/CSharpCodeParser.cs

```diff
@@ -643,12 +648,15 @@ private void SectionBlock(string left, string right, bool caseSensitive)
         private void NestedBlock()
         {
             Output(SpanKind.Code);
+
             var wasNested = IsNested;
             IsNested = true;
             using (PushSpanConfig())
             {
                 ParseBlock();
             }
+
+            Span.Start = CurrentLocation;
             Initialize(Span);
```

Comment:
Initialize is also used in some other places where we don't want to set `Span.Start`. I thought about this and didn't think it was a good plan.